wapitiCore/net/scope.py

```diff
@@ -71,3 +72,21 @@ def check(self, resource: Union[Request, str]) -> bool:
 
     def filter(self, resources: Iterable[Union[Request, str]]) -> Set[Union[Request, str]]:
         return {resource for resource in resources if self.check(resource)}
+
+
+def wildcard_translate(pattern: str) -> re.Pattern:
+    """Translate a wildcard PATTERN to a regular expression object.
+
+    This is largely inspired by fnmatch.translate.
+    """
+
+    i, length = 0, len(pattern)
+    res = ''
+    while i < length:
```

Comment:
i think `for char in pattern:` should do the job, no ?

Reply:
Indeed, that code deserved some rewriting :) Added some tests too